packages/react-aria-components/stories/Button.stories.tsx

```diff
@@ -25,6 +27,60 @@ export const ButtonExample = () => {
   );
 };
 
+export const PendingButton = {
+  render: (args) => <PendingButtonExample {...args} />,
+  args: {
+    children: 'Press me'
+  }
+};
+
+function PendingButtonExample(props) {
+  let [isPending, setPending] = useState(false);
+
+  let timeout = useRef<ReturnType<typeof setTimeout> | undefined>(undefined);
+  let handlePress = (e) => {
+    action('pressed')(e);
+    setPending(true);
+    timeout.current = setTimeout(() => {
+      setPending(false);
+      timeout.current = undefined;
+    }, 5000);
+  };
+
+  useEffect(() => {
+    return () => {
+      clearTimeout(timeout.current);
+    };
+  }, []);
+
+  let [hideOthers, setHideOthers] = useState(false);
+
+  return (
+    <>
+      {!hideOthers && (
+      <Button
+        {...props}
+        isPending={isPending}
+        onPress={handlePress}
+        className={styles2['button']}>
+          {({isPending}) => (
+            <>
+              <Text className={isPending ? styles2['pending'] : undefined}>{props.children}</Text>
+              <ProgressBar
+                aria-label="loading"
+                isIndeterminate
+                className={[styles2['spinner'], (isPending ? styles2['spinner-pending'] : '')].join(' ')}>
+                <span className={styles2['loader']} />
+              </ProgressBar>
+            </>
+          )}
+      </Button>
+    )}
+      <Button onPress={() => setHideOthers(prev => !prev)}>Toggle others</Button>
```

Comment:
What exactly is this this hide button testing for?

Reply:
Ah I see, is it to test that the `useEnforcePendingComponents` error doesn't throw if the component unmounts during pending?

Reply:
Also that it doesn't throw if unmounted in general